xarray/core/duck_array_ops.py

```diff
@@ -172,7 +166,8 @@ def isnull(data):
         )
     ):
         # these types cannot represent missing values
-        return full_like(data, dtype=bool, fill_value=False)
+        dtype = xp.bool_ if hasattr(xp, "bool_") else xp.bool
```

Comment:
I believe `xp.bool_` is only needed for compatibility with NumPy 1.x releases. Can we make that clearer here, e.g., by using `np.bool_` and adding a comment about NumPy versions?

Reply:
Will do, but also `cupy` only has `bool_` at the moment. But If I passed built-in `bool` to this I think I got a cupy error. It was a weird range from like numpy 1.2x until 2 that they deprecated `bool` and only had `bool_`, then brought it back in 2.0 :shrug: